sky/cli.py

```diff
@@ -3467,9 +3473,37 @@ def storage_delete(names: List[str], all: bool):  # pylint: disable=redefined-bu
         \b
         # Delete all storage objects.
         sky storage delete -a
+        \b
+        # Skip confirmation prompt.
+        sky storage delete -y
     """
     if sum([len(names) > 0, all]) != 1:
         raise click.UsageError('Either --all or a name must be specified.')
+    if not yes:
+        if all:
+            storages = sky.storage_ls()
+            storage_list = [s['name'] for s in storages]
+            storage_names = ', '.join(storage_list)
+            click.confirm(
+                f'Deleting {len(storage_list)} storages: '
+                f'{storage_names}. Proceed?',
+                default=True,
+                abort=True,
+                show_default=True)
+        else:
+            if len(names) > 1:
+                click.confirm(
+                    f'Deleting {len(names)} storages: '
+                    f'{", ".join(names)}. Proceed?',
+                    default=True,
+                    abort=True,
+                    show_default=True)
+            else:
+                storage_name = names[0]
+                click.confirm(f'Deleting storage: {storage_name}. Proceed?',
+                              default=True,
+                              abort=True,
+                              show_default=True)
```

Comment:
There are two major components to be achieved in this snippet from `L3482`-`L3512`. 

1. obtain the name of the storages to be deleted
2. Prompt the user if `-y` is not passed.

And these can be achieved with simpler manner if we can reorder the codes and dynamically set the message of the prompt. We can simply add a prompt once at `L3513` after obtaining the names of the storages to be deleted either with `all` or not. Then, to prompt, use something like the following after renaming the variable `names` to `storage_list`:
```
    if not yes:
        storage_names = ', '.join(storage_list)
        storage_str = 'storages' if len(storage_list) > 1 else 'storage'
        click.confirm(
            f'Deleting {len(storage_list)} {storage_str}: '
            f'{storage_names}. Proceed?',
            default=True,
            abort=True,
            show_default=True)
```

Reply:
This has been fixed!